src/Plutarch/Extra/List.hs

```diff
@@ -75,15 +89,30 @@ pmapMaybe ::
   forall (ell :: (S -> Type) -> S -> Type) (b :: S -> Type) (a :: S -> Type) (s :: S).
   (PElemConstraint ell a, PElemConstraint ell b, PListLike ell) =>
   Term s ((a :--> PMaybe b) :--> ell a :--> ell b)
-pmapMaybe = phoistAcyclic $ plam $ \f -> precList (go f) (const pnil)
+pmapMaybe = pmapMaybe'
+
+{- | A version of 'pmapMaybe' that can change the list type.
+
+     @since 3.14.1
+-}
+pmapMaybe' ::
```

Comment:
This is neat. Actually, I think it can replace `pmapMaybe`. I don't think there will be any performance overhead.

Reply:
Changing the signature of `pmapMaybe` will make existing code uncompilable. I'm not sure people are okay with this.

Reply:
I don't think this will happen: the worst it'll cause is some added type parameters to direct our choice of lists. Plus, major bump and all that - as long as you changelog it, I don't see the issue.